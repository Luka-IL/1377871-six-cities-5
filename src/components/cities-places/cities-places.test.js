import React from "react";
import renderer from "react-test-renderer";
import CitiesPlaces from "./cities-places";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";

const mockStore = configureStore([]);

const store = mockStore({
  DATA: {
    offers: [
      {
        city: {
          name: `Amsterdam`
        },
        title: `1Beautiful & luxurious studio at great location`,
        premium: false,
        type: `apartment`,
        price: 20,
        rating: 4,
        image: ``
      }, {
        city: {
          name: `Amsterdam`
        },
        title: `2Beautiful & luxurious studio at great location`,
        premium: false,
        type: `room`,
        price: 70,
        rating: 4,
        image: ``
      }
    ],
    city: `Amsterdam`
  },
  STATE: {
    sort: `popular`
  },
  USER: {
    authorizationStatus: `AUTH`
  }
});


it(`Should CitiesPlaces render correctly`, () => {
  const tree = renderer
    .create(
        <Provider store={store}>
          <CitiesPlaces
            onOfferClick ={() => {}}
          >
            <React.Fragment />
          </CitiesPlaces>
        </Provider>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
