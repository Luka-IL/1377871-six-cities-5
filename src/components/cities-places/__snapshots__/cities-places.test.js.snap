// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should CitiesPlaces render correctly 1`] = `
<section
  className="cities__places places"
>
  <h2
    className="visually-hidden"
  >
    Places
  </h2>
  <b
    className="places__found"
  >
    2
     places to stay in 
    Amsterdam
  </b>
  <form
    action="#"
    className="places__sorting"
    method="get"
  >
    <span
      className="places__sorting-caption"
    >
      Sort by
    </span>
    <select
      className="places__option places__options--custom places__sorting-type"
      id="places-sorting"
      onChange={[Function]}
    >
      <option
        className="places__option"
        value="popular"
      >
        Popular
      </option>
      <option
        className="places__option"
        value="to-high"
      >
        Price: low to high
      </option>
      <option
        className="places__option"
        value="to-low"
      >
        Price: high to low
      </option>
      <option
        className="places__option"
        value="top-rated"
      >
        Top rated first
      </option>
    </select>
  </form>
  <div
    className="cities__places-list places__list tabs__content"
  >
    <article
      className="cities__place-card place-card"
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div
        className="place-card__image-wrapper"
        onClick={[Function]}
      >
        <a
          href="#"
        >
          <img
            alt="Place image"
            className="place-card__image"
            height="200"
            src=""
            width="260"
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              20
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button button"
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              onClick={[Function]}
              width="18"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "80%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
          onClick={[Function]}
        >
          <a
            href="#"
          >
            1Beautiful & luxurious studio at great location
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          apartment
        </p>
      </div>
    </article>
    <article
      className="cities__place-card place-card"
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div
        className="place-card__image-wrapper"
        onClick={[Function]}
      >
        <a
          href="#"
        >
          <img
            alt="Place image"
            className="place-card__image"
            height="200"
            src=""
            width="260"
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              70
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button button"
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              onClick={[Function]}
              width="18"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "80%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
          onClick={[Function]}
        >
          <a
            href="#"
          >
            2Beautiful & luxurious studio at great location
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          room
        </p>
      </div>
    </article>
  </div>
</section>
`;
